lib/pure/httpclient.nim

```diff
@@ -864,6 +893,49 @@ proc newConnection(client: HttpClient | AsyncHttpClient,
     client.currentURL = url
     client.connected = true
 
+proc readFileSizes(client: HttpClient | AsyncHttpClient,
+                   multipart: MultipartData) {.multisync.} =
+  for entry in multipart.content.mitems():
+    if not entry.isFile: continue
+    if not entry.isStream:
+      entry.fileSize = entry.content.len
+      continue
+
+    when client is AsyncHttpClient:
+      var file = openAsync(entry.content)
+      let fileSize = getFileSize(file)
+      file.close()
+    else:
+      let fileSize = getFileSize(entry.content)
```

Comment:
If getting the file size blocks then AsyncFile doesn't do it properly (`getFileSize` isn't awaitable), so you may as well just get rid of this split in logic. Maybe add a TODO to look into making this async properly.